Intersect.Server.Core/Core/LogicService.LogicThread.cs

```diff
@@ -152,22 +153,25 @@ protected override void ThreadStart(ServerContext serverContext)
                             }
 
                             //Refresh list of active maps & their instances
-                            foreach (var layerId in ActiveMapInstances.Keys.ToArray())
+                            foreach (var instanceId in ActiveMapInstances.Keys.ToArray())
                             {
-                                if (!processedMapInstances.Contains(layerId))
+                                if (!processedMapInstances.Contains(instanceId))
                                 {
                                     // Remove the map entirely from the update queue
-                                    if (ActiveMapInstances[layerId] != null && ActiveMapInstances[layerId].ShouldBeCleaned())
+                                    if (ActiveMapInstances[instanceId] != null && ActiveMapInstances[instanceId].ShouldBeCleaned())
                                     {
-                                        ActiveMapInstances[layerId].RemoveLayerFromController();
-                                        ActiveMapInstances.Remove(layerId);
-                                    } else if (ActiveMapInstances[layerId] == null || !ActiveMapInstances[layerId].ShouldBeActive())
+                                        ActiveMapInstances[instanceId].RemoveLayerFromController();
+                                        ActiveMapInstances.Remove(instanceId);
+                                    } else if (ActiveMapInstances[instanceId] == null || !ActiveMapInstances[instanceId].ShouldBeActive())
                                     {
-                                        ActiveMapInstances.Remove(layerId);
+                                        ActiveMapInstances.Remove(instanceId);
                                     }
                                 }
                             }
```

Comment:
This entire thing is unsafe and unnecessarily un-cached, do this instead (or something close to it if the below has typos):

```csharp
foreach (var (instanceId, instance) in ActiveMapInstances.ToArray())
{
    if (processedMapInstances.Contains(instanceId)
    {
        continue;
    }
    
    if (instance != default)
    {
        if (instance.ShouldBeCleaned())
        {
            instance.RemoveLayerFromController();
        }
        else if (instance.ShouldBeActive())
        {
            continue;
        }
    }
    
    ActiveMapInstances.Remove(instanceId);
}
```

Reply:
Done, and also fixed a long-missing call to "MapInstance.ResetNpcSpawns();"